File: backend-spring/src/main/java/com/codequest/security/SecurityContext.java
package com.codequest.security;

public class SecurityContext {
    private static final ThreadLocal<String> userIdThreadLocal = new ThreadLocal<>();

    public static void setUserId(String userId) {
        userIdThreadLocal.set(userId);
    }

    public static String getUserId() {
        return userIdThreadLocal.get();
    }

    public static void clear() {
        userIdThreadLocal.remove();
    }
}

File: backend-spring/src/main/kotlin/com/codequest/security/SecurityContext.kt
package com.codequest.security

object SecurityContext {
    private val userIdThreadLocal = ThreadLocal<String>()

    fun setUserId(userId: String) {
        userIdThreadLocal.set(userId)
    }

    fun getUserId(): String? {
        return userIdThreadLocal.get()
    }

    fun clear() {
        userIdThreadLocal.remove()
    }
}

